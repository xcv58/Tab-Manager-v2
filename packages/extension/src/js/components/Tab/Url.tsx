import React from 'react'
import classNames from 'classnames'
import { useTheme } from 'components/hooks/useTheme'

export interface UrlProps {
  tab: {
    /**
     * The tab's URL.
     */
    url: string
  }
  /**
   * Whether the tab is duplicated.
   */
  duplicated?: boolean
  /**
   * Get the highlight node based on the query.
   */
  getHighlightNode: (url: string) => React.ReactNode
}

export default (props: UrlProps) => {
  const {
    tab: { url },
    getHighlightNode,
    duplicated,
  } = props
  const isDarkTheme = useTheme()
  return (
    <div
      className={classNames(
        'w-full overflow-hidden truncate text-xs opacity-75 group-hover:opacity-100',
        duplicated && 'text-red-200 group-hover:text-red-400',
        !duplicated && {
          'group-hover:text-black': !isDarkTheme,
          'group-hover:text-white': isDarkTheme,
        }
      )}
    >
      {getHighlightNode(url)}
    </div>
  )
}
