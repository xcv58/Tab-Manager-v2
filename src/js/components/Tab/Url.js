import React from 'react'

export default class Url extends React.Component {
  render () {
    const {
      tab: { url },
      className,
      getHighlightNode
    } = this.props
    return <div className={className}>{getHighlightNode(url)}</div>
  }
}
