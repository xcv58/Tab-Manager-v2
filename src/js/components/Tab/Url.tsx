import React from 'react'
import classNames from 'classnames'
import { useTheme } from 'components/hooks/useTheme'

export default (props) => {
  const {
    tab: { url },
    getHighlightNode,
    duplicated
  } = props
  const isDarkTheme = useTheme()
  return (
    <div
      className={classNames(
        'w-full overflow-hidden truncate text-xs opacity-75 group-hover:opacity-100',
        duplicated && 'text-red-200 group-hover:text-red-400',
        !duplicated && {
          'group-hover:text-black': !isDarkTheme,
          'group-hover:text-white': isDarkTheme
        }
      )}
    >
      {getHighlightNode(url)}
    </div>
  )
}
