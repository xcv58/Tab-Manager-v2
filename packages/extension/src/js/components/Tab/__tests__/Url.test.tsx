import React from 'react'
import { render } from '@testing-library/react'
import Url from 'components/Tab/Url'

const props = {
  tab: { url: 'url' },
  getHighlightNode: jest.fn((x) => x),
}

describe('Url', () => {
  it('render correct components', () => {
    const { container, getByText } = render(<Url {...props} />)
    expect(getByText(/url/i)).toBeInTheDocument()
    expect(getByText(/url/i)).toHaveClass(
      'w-full overflow-hidden truncate text-xs opacity-75 group-hover:opacity-100'
    )
    expect(container).toMatchSnapshot()
  })

  it('render getHighlightNode(url) as children', () => {
    const getHighlightNode = jest.fn(() => 'test')
    const { container, getByText } = render(
      <Url {...props} getHighlightNode={getHighlightNode} />
    )
    expect(getByText(/test/i)).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
