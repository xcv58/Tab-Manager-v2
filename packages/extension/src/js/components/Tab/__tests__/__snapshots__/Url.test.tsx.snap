// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Url render correct components 1`] = `
<div>
  <div
    class="w-full overflow-hidden truncate text-xs opacity-75 group-hover:opacity-100 group-hover:text-black"
  >
    url
  </div>
</div>
`;

exports[`Url render getHighlightNode(url) as children 1`] = `
<div>
  <div
    class="w-full overflow-hidden truncate text-xs opacity-75 group-hover:opacity-100 group-hover:text-black"
  >
    test
  </div>
</div>
`;
